src/Bicep.Core/Navigation/SyntaxBaseExtensions.cs

```diff
@@ -78,6 +80,42 @@ public static string ToText(this SyntaxBase syntax, string indent="")
             document.Layout(sb, indent, System.Environment.NewLine);
             return sb.ToString();
         }
+
+        /// <summary>
+        /// Generate a string that represents this Syntax element optionally preserving formatting.
+        /// </summary>
+        public static string ToTextPreserveFormatting(this SyntaxBase syntax)
```

Comment:
>ToTextPreserveFormatting [](http://example.com/codeflow?start=29&length=24)

Once of the parser baseline tests verifies that we can roundtrip from string to ProgramSyntax and back without losing formatting. Can you add an assertion that checks that calling this method on ProgramSyntax equals what you get from walking the tree with the PrintVisitor in that test?